### DISCUSSION TITLE
How does Linkerd2 tap get the body of the response?

### DISCUSSION THREAD

**Discussion by wanggithub08** [linkerd/linkerd2#8353] [Q&A]
Hey guys,I execute the command: ` linkerd viz tap -n booksapp deploy/webapp -o wide` ,and can see the response code and response length.How can Linkerd2 tap get the body of the response?

**Comment by mateiidavid** [comment]
@wanggithub08 unfortunately at the moment `tap` does not support response bodies. I noticed you opened an issue for it, we can track some design thoughts there, but presently, it won't work.

For a full list of data exposed by tap, you can have a look at the protobuf file: https://github.com/linkerd/linkerd2-proxy-api/blob/main/proto/tap.proto

**Reply by wanggithub08** [reply]
Thanks，Will you support it in the future? If I add this feature, can you give me some advice?

**Reply by kleimkuhler** [reply]
There is design that we still would need to work through for this, but since we did not have an a parent issue for this already let's move discussion there (#8359).

**Reply by wanggithub08** [reply]
ok, waiting for your good news.
